Loop/Managers/DeviceDataManager.swift

```diff
@@ -763,27 +763,28 @@ extension DeviceDataManager {
             return
         }
 
```

Comment:
This method `enactBolus` was asynchronously calling `self.loopManager.addRequestedBolus` then enacting the bolus. This indeterminate order of operations seemed to often cause this method to call the completion block before the `addRequestedBolus` which means callers can't be sure things are done upon completion. This was an extraneous issue that was leading to experiments with waits and backgroundTasks not to work.